### PR TITLE
Fix hover background on Namespace selector

### PR DESCRIPTION
**Pull request by laurakwhit**
## Description & motivation 💭 
Fix padding and margin for the `action` slot in the `Combobox` Holocene component.

### Screenshots (if applicable) 📸 
| Before | After |
|-|-|
|Screenshot 2024-07-17 at 1 58 44 PM|Screenshot 2024-07-17 at 1 58 27 PM|

### How was this tested 👻 

- [x] Manual testing
- [ ] E2E tests added
- [ ] Unit tests added

### Issue(s) closed 
`DT-2310`

### CODE REVIEW

File: src/lib/holocene/combobox/combobox.svelte [hunk 1]
```diff
@@ -301,7 +301,9 @@
     </Button>
     {#if $$slots.action}
       <div class="ml-1 h-full w-1 border-l-2 border-subtle" />
-      <slot name="action" />
+      <div class="-ml-px px-0.5">
```
Comment (rossedfort) [inline src/lib/holocene/combobox/combobox.svelte:304]: I'm not sure I love this. Can we just add `px` here?

File: src/lib/holocene/combobox/combobox.svelte [hunk 2]
```diff
@@ -301,7 +301,9 @@
     </Button>
     {#if $$slots.action}
       <div class="ml-1 h-full w-1 border-l-2 border-subtle" />
```
Comment (rossedfort) [inline src/lib/holocene/combobox/combobox.svelte:303]: Wait, why is this div closed? Is it supposed to wrap the action slot?

Comment (rossedfort) [reply src/lib/holocene/combobox/combobox.svelte:303]: Oh, maybe it's just there for the border 🤔 We can probably just add the padding to this then

File: src/lib/holocene/combobox/combobox.svelte (hunk 1 above)
Comment (laurakwhit) [reply src/lib/holocene/combobox/combobox.svelte]: Refactored [here](https://github.com/temporalio/ui/pull/2214/commits/816924bd2c8a5b4efc23b2b325b40c80dc59ecd5) ✅

File: src/lib/holocene/combobox/combobox.svelte (hunk 2 above)
Comment (laurakwhit) [reply src/lib/holocene/combobox/combobox.svelte]: That `div` was the existing border here ⬇️, but realizing I should just refactor this to one `div` 😅 Updated [here](https://github.com/temporalio/ui/pull/2214/commits/816924bd2c8a5b4efc23b2b325b40c80dc59ecd5).

Screenshot 2024-07-17 at 2 14 58 PM

**Review by rossedfort — APPROVED** [review]
🎨

**Pull request merged**